### PR TITLE
Add permissions for templates and serving runtimes

### LINKED ISSUE #1464: [Bug]: Dashboard Role doesn't have permission to create templates
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We recently migrated the Custom Serving Runtime creation to the backend API, but the dashboard' ServiceAccount doesn't have the permissions to manage the resources for `servingruntimes` and `templates` so these operations would fail.

### Expected Behavior

Users to be able to Create, Edit and Delete new Custom Serving Runtimes.

### Steps To Reproduce

1. Log in to the console
2. Try to create a new Custom Serving Runtime

### Workaround (if any)

Manually add those permissions to the odh-dashboard namespaced role.

### What browsers are you seeing the problem on?

_No response_

### Open Data Hub Version

```
Dashboard: v2.11.0
```

### Anything else

_No response_

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Closes #1464 

### Deploy the image
1. Switch to this branch
2. Change the `IMAGE_REPOSITORY` env variable in your `env.local` file to `IMAGE_REPOSITORY=quay.io/opendatahub/odh-dashboard:pr-1466`
3. Make sure you don't have the `dashboard` resource as a component in the operator `kfdef`
4. Run `make deploy`

### Test changes
1. Go to `Settings - Serving Runtimes`
2. Click “Add new”
3. Check that you can create a new Serving Runtime
4. Try to edit it
5. Try to delete it

## Test Impact

Current testing efforts are fine, this is testing dashboard service account permissions and we are excluding backend testing for now.

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] The commits have meaningful messages (squashes happen on merge by the bot).
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [X] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: manifests/base/role.yaml [hunk 1]
```diff
@@ -122,3 +122,27 @@ rules:
...
+  - apiGroups:
+      - template.openshift.io
+    verbs:
+      - get
+      - list
+      - watch
+      - create
+      - update
+      - patch
+      - delete
```
Comment (andrewballantyne) [inline manifests/base/role.yaml:134]: FWIW, you can use `*`

```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  namespace: default
  name: example.com-superuser # DO NOT USE THIS ROLE, IT IS JUST AN EXAMPLE
rules:
- apiGroups: ["example.com"]
  resources: ["*"]
  verbs: ["*"]
```
Source: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources

Comment (lucferbux) [reply manifests/base/role.yaml:134]: Yeah, you are right, I think it's better to just use the wildcard rather than listing all the verbs.

Comment (andrewballantyne) [reply manifests/base/role.yaml:134]: Since the goal is to do everything -- makes sense to use wildcard imo.

**Comment by lucferbux** [comment]
Tested with `quay.io/opendatahub/odh-dashboard:pr-1466`. I've managed to create, edit and delete the Custom Serving Runtime.

**Comment by bdattoma** [comment]
> quay.io/opendatahub/odh-dashboard:pr-1466

hey @lucferbux I used the same image on my cluster, but it keeps failing with the same 403 error. Does the image contain the necessary Role?

**Comment by pnaik1** [comment]
working fine for me!!
[Screencast from 07-05-2023 04:22:36 PM.webm](https://github.com/opendatahub-io/odh-dashboard/assets/99238909/f604671f-989a-4c65-98aa-9c6a785ed27b)

**Comment by lucferbux** [comment]
> > quay.io/opendatahub/odh-dashboard:pr-1466
> 
> hey @lucferbux I used the same image on my cluster, but it keeps failing with the same 403 error. Does the image contain the necessary Role?
> 
> The Role is not created in my cluster when after i replaced the RHODS Dashboard image with the pr-1466 one. I guess it's expected right? Would I need the operator to apply the Role?

Hi berto, no, it shouldn't work, as the image itself doesn't contain the changes, that's why we use the whole **make deploy**. The change itself is targeted for the `odh-dashboard` role. You can edit that role with this change to see if it's working. I'm gonna cherry-pick these changes into odh-manifest downstream once it's merged.

**Comment by lucferbux** [comment]
> working fine for me!! [Screencast from 07-05-2023 04:22:36 PM.webm](https://github.com/opendatahub-io/odh-dashboard/assets/99238909/f604671f-989a-4c65-98aa-9c6a785ed27b)

@pnaik1 Perfect, can you lgtm?

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

**Comment by lucferbux** [comment]
Fixed issue in the `migrateTemplateDisablement` which was preventing the dashboardconfig to be fetched by the watcher.

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

**Pull request merged**